prboom2/src/SDL/i_system.c

```diff
@@ -315,7 +319,13 @@ const char *I_ConfigDir(void)
     const char *home = M_getenv("HOME");
     if (!home)
     {
-      home = "/";
+#ifdef HAVE_GETPWUID
+      struct passwd *user_info = getpwuid(getuid());
+      if (user_info != NULL)
+        home = user_info->pw_dir;
```

Comment:
Not sure if the struct pointer that `getpwuid()` returns remains static in memory. Any other process calling `getpwuid()` might cause its memory to change, so it should be safer to call `strdup()` on the `pw_dir` string.

Reply:
According to the man page:
```
The return value may point to a static area, and may be overwritten
by subsequent calls to getpwent(3), getpwnam(), or getpwuid().
(Do not pass the returned pointer to free(3).)
```
Unless there are multiple threads running which may call this function, this should be safe. Once the `base` static variable is initialized, this won't get called again.